zookeeper-server/src/test/java/org/apache/zookeeper/test/ZkDatabaseCorruptionTest.java

```diff
@@ -196,4 +198,13 @@ public File findMostRecentSnapshot() throws IOException {
         assertEquals(0, zkDatabase.calculateTxnLogSizeLimit());
     }
 
+    @Test
+    public void testCommittedLogSizeLimit() throws IOException {
+        ZKDatabase zkDatabase = new ZKDatabase(new FileTxnSnapLog(new File("foo"), new File("bar")));
```

Comment:
We could set a very low limit here and assert that the proposal is removed when the limit is reached.